stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamImpl.java

```diff
@@ -115,34 +124,37 @@ public KStreamImpl(TopologyBuilder topology, String name) {
     public <V1> KStream<K, V1> flatMapValues(ValueMapper<V, ? extends Iterable<V1>> mapper) {
         String name = FLATMAPVALUES_NAME + INDEX.getAndIncrement();
 
-        topology.addProcessor(name, KStreamFlatMapValues.class, new ProcessorMetadata("ValueMapper", mapper), this.name);
+        topology.addProcessor(name, new KStreamFlatMapValues(mapper), this.name);
 
         return new KStreamImpl<>(topology, name);
     }
 
     @Override
     public KStreamWindowed<K, V> with(Window<K, V> window) {
-        KStreamWindow<K, V> windowed = new KStreamWindow<>(window);
+        String name = WINDOWED_NAME + INDEX.getAndIncrement();
 
-        topology.addProcessor(windowed, processor);
+        topology.addProcessor(name, new KStreamWindow<>(window), this.name);
 
-        return new KStreamWindow.KStreamWindowedImpl<>(topology, windowed);
+        return new KStreamWindowedImpl<>(topology, name, window);
     }
 
     @Override
     @SuppressWarnings("unchecked")
     public KStream<K, V>[] branch(Predicate<K, V>... predicates) {
-        String name = BRANCH_NAME + INDEX.getAndIncrement();
+        String branchName = BRANCH_NAME + INDEX.getAndIncrement();
 
-        topology.addProcessor(name, KStreamBranch.class, new ProcessorMetadata("Predicates", Arrays.copyOf(predicates, predicates.length)), this.name);
+        topology.addProcessor(branchName, new KStreamBranch(predicates.clone()), this.name);
```

Comment:
Since we have removed ProcessorMetadata, could we directly pass the arbitrary-number parameter here?